### DISCUSSION TITLE
Mark AI with their reaction time(s)

### DISCUSSION THREAD

**Discussion by Skilly146** [Tsangares/stoned_ape_tools#11] [Feature Requests]
## What

Pretty simple idea, just put a countdown to when the AI will next react or even just one of the times it will react since they react every 4 hours after they become AI's.

### How

There is plenty of space next to a players name to add the times, so the only challenge I see is how to find when the AI will react since you can't rely on being able to figure it out during the small period the extension is active. Fortunately an AI's reaction time is always based off of when they were converted (I don't know exactly when, either 1 tick after, 4 ticks after, or immediately) so if the extension can just look through the events they can find the tick that the player quit or went afk. Since it seems like the ledger already uses a similar method it should be pretty easy to adapt

**Comment by Skilly146** [comment]
I just went into a test game and figured out the specifics. Bots begin reacting 3 ticks after they become a bot and every 4 ticks afterwards. Example:

Bot leaves at tick 13. It reacts for the first time at tick 16 (which can be seen on tick 17) and reacts every fourth tick afterwards, so ticks 20, 24, 28 (seen 21, 25, 29) and so on'

The easiest way to figure out a bots reaction times would be the formula `(tickBotLeft - 1) * multiplesOfFour` or to find if the current tick is a reaction tick you go ` if (currentTick - (tickBotLeft - 1) % 4) == 0`

**Reply by Tsangares** [reply]
You are a genius. You do most of the hard work when doing things like this. Then I just have to type it in and make it look petty. Thank you
